vaadin-radio-button-flow-demo/src/main/java/com/vaadin/flow/component/radiobutton/demo/RadioButtonGroupView.java

```diff
@@ -15,330 +15,287 @@
-        group.addValueChangeListener(event -> message.setText(String.format(
-                "Radio button group value changed from '%s' to '%s'",
-                getName(event.getOldValue()), getName(event.getValue()))));
+        // source-example-heading: Read-only
+        RadioButtonGroup<String> radioGroup = new RadioButtonGroup<>();
+        radioGroup.setItems("Option one", "Option two", "Option three");
+        radioGroup.addThemeVariants(RadioGroupVariant.LUMO_VERTICAL);
+        radioGroup.setValue("Option one");
+        radioGroup.setReadOnly(true);
         // end-source-example
 
-        group.setId("button-group-renderer");
-        message.setId("button-group-renderer-value");
-
-        addCard("Radio button group with renderer", group, message);
+        addCard("Read-only", radioGroup);
     }
 
-    private void addItemLabelGenerator() {
-        Div message = new Div();
-
-        // begin-source-example
-        // source-example-heading: Radio button group with label generator
-        RadioButtonGroup<Person> group = new RadioButtonGroup<>();
-        group.setItems(new Person("Joe"), new Person("John"),
-                new Person("Bill"));
-        group.setRenderer(new TextRenderer<>(Person::getName));
-        group.addValueChangeListener(event -> message.setText(String.format(
-                "Radio button group value changed from '%s' to '%s'",
-                getName(event.getOldValue()), getName(event.getValue()))));
-        // end-source-example
-
-        group.setId("button-group-with-item-generator");
-        message.setId("button-group-gen-value");
+    private List<Department> getDepartments() {
 
-        addCard("Radio button group with label generator", group, message);
+        DepartmentData departmentData = new DepartmentData();
+        return departmentData.getDepartments();
     }
 
-    private void addItemIconGenerator() {
+    private void entityList() {
         // begin-source-example
-        // source-example-heading: Radio button group with icon generator
-        RadioButtonGroup<Person> group = new RadioButtonGroup<>();
-        group.setItems(new Person(1, "Joe"), new Person(2, "John"),
-                new Person(3, "Bill"));
-        group.setRenderer(new IconRenderer<>(item -> {
-            Image image = new Image("https://vaadin.com/images/vaadin-logo.svg",
-                    "");
-            image.getStyle().set("height", "15px");
-            image.getStyle().set("float", "left");
-            image.getStyle().set("marginRight", "5px");
-            image.getStyle().set("marginTop", "2px");
-            return image;
-        }, Person::getName));
+        // source-example-heading: Entity list
+        RadioButtonGroup<Department> radioGroup = new RadioButtonGroup<>();
+        radioGroup.setLabel("Department");
+        List<Department> departmentList = getDepartments();
+        radioGroup.setItems(departmentList);
```

Comment:
The demo should explicitly set the visual representation of a Department. I believe it's currently relying on the entity's `toString()` which isn't really a best practice. Add something like `radioGroup.setRenderer(new TextRenderer<>(Department::getName));`

Reply:
Done.